create_mode.go

```diff
@@ -0,0 +1,52 @@
+package zk
+
+import "fmt"
+
+const (
+	FlagPersistent                  = 0
+	FlagEphemeral                   = 1
+	FlagSequence                    = 2
+	FlagEphemeralSequential         = 3
+	FlagContainer                   = 4
+	FlagTTL                         = 5
+	FlagPersistentSequentialWithTTL = 6
+)
+
+type createMode struct {
+	flag        int32
+	ephemeral   bool
+	sequential  bool
+	isContainer bool
+	isTTL       bool
+}
+
+func (cm *createMode) toFlag() int32 {
```

Comment:
optional/nit: not sure if we need a getter for a private struct, can just use the field directly